include/rocksdb/env.h

```diff
@@ -422,6 +425,23 @@ class Env {
   // When "function(arg)" returns, the thread will be destroyed.
   virtual void StartThread(void (*function)(void* arg), void* arg) = 0;
 
+  // Start a new thread, invoking "function(args...)" within the new thread.
+  // When "function(args...)" returns, the thread will be destroyed.
+  template <typename FunctionT, typename... Args>
+  auto WrapStartThread(FunctionT function, Args&&... args) ->
```

Comment:
Being "wrapped" is an implementation detail. The key feature vs. `StartThread` is type safety. Perhaps `StartThreadTyped`